src/Grpc/GrpcTransportTrait.php

```diff
@@ -173,7 +145,46 @@ public function createStub($generatedCreateStub, $args = [])
         if (isset($args['forceNewChannel']) && $args['forceNewChannel']) {
             $stubOpts['force_new'] = true;
         }
-        return $generatedCreateStub($fullAddress, $stubOpts, $channel);
+
+        if (empty($args['createGrpcStubFunction'])) {
```

Comment:
I added this to help with unit tests, but I am not sold on it. Also the whole `$grpcStubClassName` is pretty weak. Also it may be best to make this an abstract base class rather than a trait.